data/tutorials/language/0it_01_basic_datatypes.md

```diff
@@ -482,6 +482,16 @@ Note that:
 - `unit` is a variant with a unique constructor, which does not carry data: `()`.
 - `bool` is also a variant with two constructors that doesn't carry data: `true` and `false`.
 
+#### Empty Variants
+
+Variants can optionally be defined with absolutely no constructors at all.
+
+```ocaml
+type void = |
+```
+
+Such types are not ordinarily useful in OCaml programs (as they do not have any constructible values), but they can be useful as temporary placeholders when defining types in a [functor](/docs/functors#writing-your-own-functors).
+
```

Comment:
This is not the intended use case for empty types. I would avoid mentioning them before the GADT introduction.

Reply:
I see. That would make sense. I'll revert the corresponding commit.

Reply:
I've edited the `Binary` functor example to make `elt` and `t` abstract types.

Thanks for the link to the manual! I'll check that out.

Reply:
To give a better example of what I meant by "abstract type" are better placeholder, if you write
```ocaml
type empty = |
let f ([]: empty list) = ()
```
the function `f` is total: the only list of empty values is the empty list. But this is property is only true for empty types. In other words, by using empty types as placeholders one might accidentally capture properties that are only true for empty types.

Reply:
I see. I think I better understand now. As a an aside, are there any examples in the wild where using empty types and refutation cases for them come in handy when it comes to GADTs? I think I'm still fuzzy on the motivation for empty types.